Formula/lib/libxaw3d.rb

```diff
@@ -23,19 +24,26 @@ class Libxaw3d < Formula
   depends_on "libxpm"
   depends_on "libxt"
 
+  # Backport fix for improved linking on macOS
+  # Patch doesn't apply on Monterey so only fix newer macOS
```

Comment:
How can a patch only apply selectively? That's weird.

Reply:
No idea. Was failing on Linux, so moved to macOS-only. Then failed only on Monterey - https://github.com/Homebrew/homebrew-core/actions/runs/10549634844/job/29224919518#step:3:524

Could be the version of `patch`.

Reply:
Now checking code, curious how patch worked on Ventura/Sonoma.

For `libxt`, would need https://gitlab.freedesktop.org/xorg/lib/libxt/-/commit/a938ec5ca1923759327b7b80861d8382b3686a1b. Somehow macOS patch was able to ignore the removed `#if`.

Maybe similar on other formulae.